consensus/consensus-types/src/timeout_certificate.rs

```diff
@@ -39,11 +43,17 @@ impl TimeoutCertificate {
         }
     }
 
-    /// Verifies the signatures for the round
+    /// Verifies the signatures for each validator, it signs on the timeout round and the highest qc round.
     pub fn verify(&self, validator: &ValidatorVerifier) -> anyhow::Result<()> {
-        validator
-            .verify_aggregated_struct_signature(&self.timeout, &self.signatures)
-            .context("Failed to verify TimeoutCertificate")?;
+        // Verify enough signatures
+        validator.check_voting_power(self.signatures.keys())?;
+        // Verify each node's timeout attestation.
+        for (author, (hqc_round, signature)) in &self.signatures {
+            let t = Timeout::new(self.timeout.epoch(), self.timeout.round(), *hqc_round);
```

Comment:
I probably miss something. Why can't you use self.timeout. hqc_round() instead of hqc_round?